src/components/List.jsx

```diff
@@ -0,0 +1,30 @@
+import React, { useEffect, useState } from "react";
+import { getList } from "../lib/api/get";
+import { useParams } from "react-router-dom";
+
+const List = (props) => {
+  const [data, setData] = useState([]);
+
+  const query = useParams();
+
+  useEffect(() => {
+    handleGetDetail(query);
+  }, [query]);
+
+  const handleGetDetail = async (query) => {
+    try {
+      const res = await getList(query.id);
+      console.log(res.data);
+      setData(res.data);
+    } catch (e) {
+      console.log(e);
+    }
+  };
+
+  return (
+    <div>
+      <h1>Microposts</h1>
```

Comment:
とりあえずまずは
```suggestion
      <h1>Microposts</h1>
      <div>{JSON.stringify(data)}</div>
```
に変えたとき表示がどうなるか見てみましょうか。

Reply:
<img width="1509" alt="スクリーンショット 2022-04-25 17 28 42" src="https://user-images.githubusercontent.com/95409537/165050809-eeefa010-6272-4128-9b5e-0e2e1af8db51.png">

コンソールで表示していたjsonが表示されました。

Reply:
これ、API 側の実装がちょっとバグってますね。

欲しいのは `[{ microposts: [...] }]` じゃなくて `{ microposts: [...] }` のハズ。

Reply:
それはまあ直してもらうとして、
HTML 内に埋め込めたので、あとは client 側のレイアウトを調整していきましょう
```javascript
// 一部省略あり
const List = () => {
  const params = useParams();
  const [userMicroposts, setUserMicroposts] = useState([]);

  useEffect(async () => {
    try {
      const data = await getList();
      setUserMicroposts(data.microposts);
    } catch (e) {
      ...
    }
  });

  // <li>...</li> を好きに弄ってレイアウトを作ろう！サブコンポーネントを作るのがきれいだと思うよ
  return (
    <div>
      <h1>Microposts</h1>
      <ol>
        {microposts.map(mp => <li key={mp.id}>mp.id</li>)}
      </ol>
    </div>
  );
};
```

Reply:
APIのバグについて、これで正しい表示でしょうか？
<img width="1512" alt="スクリーンショット 2022-04-26 9 39 09" src="https://user-images.githubusercontent.com/95409537/165196536-a4f0e59e-2cbd-4be7-9cda-07938c1e1691.png">

Reply:
僕が何を元に「バグだ」と指摘したかというと、 @johnsmith1202 の書いてくれたこの仕様書と合ってなかったからです。
「僕の気分」とかではないので、「僕に聞かないと合ってるかわからない」みたいな聞き方をされるのは「？」となってしまいます

https://github.com/sls-training/sample_app/blob/aa4e578598cc7bdfc83b36dbb817392c7b8f4f67/API%E8%A8%AD%E8%A8%88%E6%9B%B8.md#%E3%83%AC%E3%82%B9%E3%83%9D%E3%83%B3%E3%82%B9%E3%82%B5%E3%83%B3%E3%83%97%E3%83%AB

Reply:
「どう実装するとよいか」のアドバイスは出来ますが、
「どういった仕様であるべきか」については、ここでは僕が決めることではないと思っているので、そのつもりでコメントしていくのでよろしく。

Reply:
なるほど。
「仕様書と違う」という意味ではなく、「そもそもjsonの形として違う」という意味で受け取っていました。